great_expectations/data_context/types/__init__.py

```diff
@@ -1 +1,38 @@
-from .configurations import ClassConfig
\ No newline at end of file
+from ...types import LooselyTypedDotDict
+from six import string_types
+from collections import namedtuple
+from six import string_types
+
+from .configurations import (
+    Config,
+    ClassConfig,
+    DataContextConfig,
+)
+
+NormalizedDataAssetName = namedtuple("NormalizedDataAssetName", [
+    "datasource",
+    "generator",
+    "generator_asset"
+])
+
+
+class NameSpaceDotDict(LooselyTypedDotDict):
```

Comment:
We can change this later, but I don't think we want to *require* all of those keys.

Reply:
Yes, I agree. This `NameSpaceDotDict` is still quite wippy.

Do you think it'd be worth making a PR to implement the classes here? : https://docs.google.com/document/d/1PALba5-D83oa3RmJOGi2DObnSFCe3CICqIkwCo2fHqw/edit

I think the goal would be to
1. define the classes in a first PR, and
2. subsume NameSpaceDotDict into ValidationResultSuiteIdentifier in a second.

From there we could progressively roll out the other types in targeted refactor PRS.

Reply:
wfm (but so does just doing this incrementally)